lib/legacy/polymer.dom.html

```diff
@@ -235,6 +235,64 @@
     }
   }
 
+  /**
+   * @function
+   * @param {boolean=} deep
+   * @return {!Node}
+   */
+  Polymer.DomApi.prototype.cloneNode;
+  /**
+   * @function
+   * @param {!Node} node
+   * @return {!Node}
+   */
+  Polymer.DomApi.prototype.appendChild;
+  /**
+   * @function
+   * @param {!Node} newChild
+   * @param {Node} refChild
+   * @return {!Node}
+   */
+  Polymer.DomApi.prototype.insertBefore;
+  /**
+   * @function
+   * @param {!Node} node
+   * @return {!Node}
+   */
+  Polymer.DomApi.prototype.removeChild;
+  /**
+   * @function
+   * @param {!Node} oldChild
+   * @param {!Node} newChild
+   * @return {!Node}
+   */
+  Polymer.DomApi.prototype.replaceChild;
+  /**
+   * @function
+   * @param {string} name
+   * @param {string} value
+   * @return {void}
+   */
+  Polymer.DomApi.prototype.setAttribute;
+  /**
+   * @function
+   * @param {string} name
+   * @return {void}
+   */
+  Polymer.DomApi.prototype.removeAttribute;
+  /**
+   * @function
+   * @param {string} selector
+   * @return {?Element}
+   */
+  Polymer.DomApi.prototype.querySelector;
+  /**
+   * @function
+   * @param {string} selector
+   * @return {!Array<!Element>}
+   */
+  Polymer.DomApi.prototype.querySelectorAll;
```

Comment:
this should be a `NodeList` like defined [here](https://github.com/google/closure-compiler/blob/032a257f7ec86effe9171736b138dae0f3d9d86a/externs/browser/w3c_dom1.js#L436). what should we do about this? it shouldn't be an `Array`

Reply:
I see `NodeList` in the TypeScript standard lib: https://github.com/Microsoft/TypeScript/blob/master/lib/lib.d.ts#L12654 (warning: huge file)

Reply:
im having trouble with this because it should be (in TS) `NodeListOf<Element>`, but in closure they seem to use `NodeList<Element>`. either way, i just end up with `any` unless i pass _just_ `NodeList` (without a `T`) :/

Reply:
Ah, I see. `gen-typescript-declarations` doesn't currently handle the general case of a type applications `Foo<Bar>`, it only handles the special cases `Array<Foo>` and `Object<Foo, Bar>`. I just filed https://github.com/Polymer/gen-typescript-declarations/issues/58 to track that.

We'll also need some special logic for converting `NodeList<Foo>` to `NodeListOf<Foo>`, since TypeScript has two different types for when it is parameterized vs not. Filed https://github.com/Polymer/gen-typescript-declarations/issues/59 for that. Although, I do wonder if the TypeScript types could be updated so that `NodeList` has a default generic type parameter of `Node` (which is possible as of TypeScript 2.3: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-3.html#generic-parameter-defaults).

Let's not block merging this PR on fixing those, though. It looks like TypeScript's `NodeList` effectively has `Node` as the default type, so if Closure is ok with a non-parameterized `NodeList`, then that seems ok, even if we're sure that this can only return `Elements` -- definitely better than `Array`.